go/libraries/doltcore/env/environment.go

```diff
@@ -1251,17 +1254,71 @@ func (dEnv *DoltEnv) LockFile() string {
 	return f
 }
 
-// IsLocked returns true if this database's lockfile exists
+// IsLocked returns true if this database's lockfile exists and the pid contained in lockfile is alive.
 func (dEnv *DoltEnv) IsLocked() bool {
 	if dEnv.IgnoreLockFile {
 		return false
 	}
 
 	ok, _ := dEnv.FS.Exists(dEnv.LockFile())
-	return ok
+	if !ok {
+		return false
+	}
+
+	lockFilePid, err := getProcessFromLockFile(dEnv.FS, dEnv.LockFile())
+	if err != nil { // if there's any error assume that env is locked since the file exists
+		return true
+	}
+
+	// Check whether the pid that spawned the lock file is still running. Ignore it if not.
+	p, err := os.FindProcess(lockFilePid)
+	if err != nil { // if there's any error assume that env is locked since the file exists
+		return true
+	}
+
+	if p == nil {
+		return false
+	}
+
+	return checkPid(p.Pid)
 }
 
-// Lock writes this database's lockfile or errors if it already exists
+// cc. https://stackoverflow.com/questions/15204162/check-if-a-process-exists-in-go-way
+func checkPid(pid int) bool {
+	out, err := exec.Command("kill", "-s", "0", strconv.Itoa(pid)).CombinedOutput()
```

Comment:
suspicious of whether this will work with windows

Reply:
I added some notes

Reply:
It for sure will not work

Reply:
You should probably just not do this check on windows